services/src/api/handlers/datasets.rs

```diff
@@ -220,6 +221,58 @@ pub async fn get_dataset_handler<C: ApplicationContext>(
     Ok(web::Json(dataset))
 }
 
+/// Update details about a dataset using the internal name.
+#[utoipa::path(
+    tag = "Datasets",
+    post,
```

Comment:
Isn't this again put because of idempotency?

Reply:
it's not idempotent unfortunately. If you have dataset named "A" and you rename it to "B", trying to rename it again will fail because dataset A does no longer exist.

something like this would happen:
> POST /datastet/A 
> {
>  "name": "B"
> }
> 200 OK
>
> POST /datastet/A 
> {
>  "name": "B"
> }
> 404 NOT FOUND